Oop-1 Question2 Matrix/Matrix.java

```diff
@@ -0,0 +1,184 @@
+			for (int columnIndex = 0; columnIndex < noOfRow; columnIndex++) {
+				tranposeMatrix[rowIndex][columnIndex] = matrixElement[columnIndex][rowIndex];
+			}
+		}
+		return tranposeMatrix;
+	}
+	
+	/**
+	 *  method to show the matrix elements
+	 * @return a string of matrix elements separated by tabs and newline 
+	 */
+	public String show() {
+		if(matrixElement==null) {
+			throw new NullPointerException();
+		}
+		String matrix = "";
+		
+		for (int rowIndex = 0; rowIndex < noOfRow; rowIndex++) {
+			
+			for (int columnIndex = 0; columnIndex < noOfColumn; columnIndex++) {
+				matrix = matrix + matrixElement[rowIndex][columnIndex] + "\t";
+			}
+			
+			matrix = matrix + "\n";
+			
+		}
+		return matrix;
+	}
+	
+	/**
+	 * This method is designed to multiply two matrices
+	 * @param multiplierMatrix the matrix to be multiplied with calling matrix
+	 * @return the multiplication matrix of the passed matrices
+	 * @throws IllegalArgumentException
+	 * @throws NullPointerException
+	 */
+	public Matrix multiplication(Matrix multiplierMatrix)
+			throws IllegalArgumentException, NullPointerException {
+		Matrix multiplicationMatrix;
+		int sum;
+		if(multiplierMatrix==null) {
+			
+			throw new NullPointerException();
+			
+		}
+		
+		if (noOfColumn == multiplierMatrix.getNoOfRow()) {
+
+			multiplicationMatrix = new Matrix(noOfRow,
+					multiplierMatrix.getNoOfColumn());
+
+			for (int rowIndex = 0; rowIndex < getNoOfRow(); rowIndex++) {
+
+				for (int columnIndex = 0; columnIndex < multiplierMatrix.getNoOfColumn(); columnIndex++) {
+
+					sum = 0;
+					for (int internalIndex = 0; internalIndex < getNoOfColumn(); internalIndex++) {
+
+						sum = sum
+								+ (this.matrixElement[rowIndex][internalIndex] 
+										* multiplierMatrix.getElement(internalIndex,columnIndex));
+
+					}
+
+					multiplicationMatrix.setElement(rowIndex, columnIndex, sum);
+
+				}
+
+			}
+
+		} else {
+			throw new IllegalArgumentException("Dimensions of matrix not suitable for multiplication");
+		}
+		return multiplicationMatrix;
+	}
+	
+	/**
+	 * This method is designed to add two matrices
+	 * @param addendMatrix the matrix to be added with the calling matrix
+	 * @return the addition matrix of the passed matrices
+	 * @throws IllegalArgumentException
+	 * @throws NullPointerException
+	 */
+	public Matrix addMatrix(Matrix addendMatrix) throws IllegalArgumentException,NullPointerException {
+		Matrix additionMatrix;
+		int sum;
+		
+		if(addendMatrix==null) {
+			
+			throw new NullPointerException();
+			
+		}
+
+		if (noOfRow == addendMatrix.getNoOfRow() && noOfColumn==addendMatrix.getNoOfColumn()) {
+
+			additionMatrix = new Matrix(noOfRow,noOfColumn);
+
+			for (int rowIndex = 0; rowIndex < addendMatrix.getNoOfRow(); rowIndex++) {
+
+				for (int columnIndex = 0; columnIndex < addendMatrix
+						.getNoOfColumn(); columnIndex++) {
+
+					sum = 0;
+					sum=this.matrixElement[rowIndex][columnIndex]+addendMatrix.getElement(rowIndex,columnIndex);
```

Comment:
can we do like this:
additionMatrix.setElement(rowIndex, columnIndex, this.matrixElement[rowIndex][columnIndex]+addendMatrix.getElement(rowIndex,columnIndex));
  just to reduce variable declaration